sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/DerivationToolExt.kt

```diff
@@ -52,6 +54,19 @@ fun Derivation.deriveUnifiedFullViewingKeysTypesafe(
         numberOfAccounts
     ).map { UnifiedFullViewingKey(it) }
 
+fun Derivation.deriveAccountMetadataKeyTypesafe(
```

Comment:
The naming convention in this file is not consistent: some methods have the `Typesafe` suffix and some do not. There is absolutely no difference in type safety between the methods in this interface that have `Typesafe` in their names and the ones that do not.

Also, there is a useless `TypesafeDerivationToolImpl` wrapper that is literally doing nothing but delegate to an object of type `Derivation`, even though `Derivation` is *already* a perfectly fine typesafe interface.

I know this PR is not trying to fix this, but please can we avoid making it worse? Delete `Typesafe` from the names of new methods, so that it'll be simpler to just strip out `TypesafeDerivationToolImpl` in a follow-up PR.